auth-frontend/src/components/Login.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import AuthService from '../configs/authService';
+
+class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: '',
+      password: '',
+    };
+    this.service = new AuthService();
+  }
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  };
+
+  handleFormSubmit = (event) => {
+    event.preventDefault();
+    const username = this.state.username;
+    const password = this.state.password;
+    this.service
+      .login(username, password)
+      .then((response) => {
+        this.setState({ username: '', password: '' });
```

Comment:
this.setState es asyncrono, por lo cual si vas a hacer un push despues, para que no intente hacer un setState con el componente desmontado podes hacer

```
this.setState({...}, () => {
        this.props.getTheUser(response);
        this.props.history.push('/profile');
})
```